app/components/post_list/post/header/custom_status/index.tsx

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {Keyboard, TouchableOpacity} from 'react-native';
+
+import CustomStatusEmoji from '@app/components/custom_status/custom_status_emoji';
+import {Screens} from '@app/constants';
+import {openAsBottomSheet} from '@app/screens/navigation';
+import {makeStyleSheetFromTheme} from '@app/utils/theme';
+import {preventDoubleTap} from '@utils/tap';
+
+type HeaderCustomStatusEmojiProps = {
+    channelId: string;
+    location: string;
+    theme: Theme;
+    userIconOverride?: string;
+    usernameOverride?: string;
+    userId: string;
+    customStatus: UserCustomStatus;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        customStatusEmoji: {
+            color: theme.centerChannelColor,
+            marginRight: 4,
+            marginTop: 2,
+        },
+    };
+});
+
+const HeaderCustomStatusEmoji = ({
```

Comment:
This looks fine, but it is almost the same as the `HeaderDisplayName` component. I wonder if it would be possible to extract the Touchable outside of the component, and it to cover both the display name and the status in one single touchable.

Are you open to try that? It will need a bit of refactoring of the `HeaderDisplayName` component (or may not even be needed anymore).

Reply:
ok will do that

Reply:
Instead of shifting the logic of Touchable into the parent component `Header`, can we shift the `CustomStatusEmoji` component into `HeaderDisplayName`?

Reply:
On the paper, looks good to me. Let's see how the code looks :smile: